brand_external_report_layout/views/report_template.xml

```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2019 ACSONE SA/NV
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+
+<odoo>
+
+    <template id="brand_external_layout" inherit_id="web.external_layout">
+        <xpath expr="//t[@t-if='not company']" position="after">
+            <t t-if="'brand_id' in o.fields_get() and o.brand_id and o.brand_id.external_report_layout_id"
+               t-call="{{o.brand_id.external_report_layout_id.key}}">
+                <t t-set="company" t-value="o.brand_id.sudo()"/>
```

Comment:
So yes, this is a little bit of a hack, but I think it's worth it for the sake of code simplicity.
Knowing that reports normally don't use this field and rely on the record `company_id` field, this should not be an issue in practice, as explained in the README.